Problem 2.cpp

```diff
@@ -0,0 +1,30 @@
+#include <iostream>
+#include <algorithm>
+#include <vector>
+using namespace std;
+
+int main()
+{
+    int x1, x2, x3, x4;
+    cin >> x1;
+    cin >> x2;
+    cin >> x3;
+    cin >> x4;
+    
+    int minx, maxx;
+
+    cout << (minx = min(min(x1, x2), min(x3, x4)));
```

Comment:
You seem to get the general idea, and in fact this is a cool way to only do log(n) many mins and maxes. However, what if I had asked you to do this for 1000 numbers?
I suggest you use an array for this part. 
In the problem I said to use `min` and `max`, but you could use `min_element` and `max_element` on the array :)